### ISSUE TITLE
feat: Ability to no clear the search box when clicked outside

### ISSUE THREAD

**Issue by erezrokah** [ghoshnirmalya/react-search-box#170]
Thank you for this lovely component 🚀 

At the moment when you click outside the search box the test gets clear:
https://github.com/ghoshnirmalya/react-search-box/blob/af881e89d6091c06aae11bd7ff0d129bf7026fb6/src/hooks/use-outside-click.ts#L12

Are you option to add a configuration to control that behavior? Some users might prefer to keep the text they typed

**Comment by ghoshnirmalya** [comment]
@erezrokah Yes. That sounds like a reasonable request. Are you interested in adding that functionality?

**Comment by erezrokah** [comment]
Sure I can submit PR. Thanks @ghoshnirmalya!

**Comment by chstavridis** [comment]
What is the status of this issue?

**Comment by erezrokah** [comment]
Sorry @chstavridis, that's on me I haven't been able to submit a PR yet. We'll try to do it this week

**Comment by alymkarimg** [comment]
hi guys, any way to fix this!

**Comment by ghoshnirmalya** [comment]
@alymkarimg I noticed that you created [a PR](https://github.com/ghoshnirmalya/react-search-box/pull/180) and then closed it. Would you like to build that functionality? I can help.

**Comment by alymkarimg** [comment]
Hi, Yh, I’d love to help! I’ve also got one more bug that can’t be fixed
unless I change the search box to an uncontrolled component, would you be
free for a video call anytime in the near future!

Kind regards
Alym Karim
